lib/guardian/topic_guardian.rb

```diff
@@ -145,22 +145,23 @@ def is_in_edit_topic_groups?
   end
 
   def can_recover_topic?(topic)
-    if is_staff? || (topic&.category && is_category_group_moderator?(topic.category)) ||
+    return false if topic.blank?
```

Comment:
Hoisted this check up here because it was checked in all three expressions below.